plugin/src/main/java/org/openrewrite/gradle/isolated/DefaultProjectParser.java

```diff
@@ -737,7 +739,7 @@ public Stream<SourceFile> parse(Project subproject, Set<Path> alreadyParsed, Exe
                             .flatMap(jp -> jp.parse(javaPaths, baseDir, ctx))
                             .map(cu -> {
                                 if (isExcluded(exclusions, cu.getSourcePath()) ||
-                                    cu.getSourcePath().startsWith(baseDir.relativize(subproject.getLayout().getBuildDirectory().get().getAsFile().toPath()))) {
+                                        cu.getSourcePath().startsWith(buildDirPath)) {
```

Comment:
Do we want to even be parsing these files at all with OpenRewrite?

Reply:
I thought we did so for the type attribution; that's at least [the comment over on the Maven plugin](https://github.com/openrewrite/rewrite-maven-plugin/blob/ecf91b17f3dfe11cf35fa10e8c448651654497e5/src/main/java/org/openrewrite/maven/MavenMojoProjectParser.java#L387-L393). Hadn't thought to change that here just yet. Just excluding them after they are parsed to prevent them ending up in the LST themselves.

Reply:
So we have so far, yeah, but we also trigger the compile phases in Gradle as well prior to the rewrite tasks, so technically we could put the source set's own output classes onto the parser's classpath to still get the necessary type attribution and bypass parsing the generated types.

Just a thought though.

Reply:
Could indeed be a nice improvement to stop ourselves from parsing source files whose LSTs we'll discard anyway, or perhaps even having two sources of the same type information; compiled vs source.